app/Models/Log.php

```diff
@@ -52,6 +52,51 @@ public function setLogDataAttribute($value)
         $this->attributes['log_data'] = serialize($value);
     }
 
+    public function dataForDisplay()
+    {
+        if ($this->log_operation === 'LOG_TOPIC_TYPE') {
+            switch ($this->log_data['type']) {
+                case 0:
+                    $translationString = 'unpin';
+                    break;
+                case 1:
+                    $translationString = 'pin';
+                    break;
+                case 2:
+                    $translationString = 'announcement';
+                    break;
+            }
+
+            return trans("forum.topic.logs.data.{$translationString}");
+        }
+
+        $data = $this->log_data[0];
+        $translationKey = $this->translationKey();
+
+        $noTranslation = ['LOG_DELETE_TOPIC', 'LOG_LOCK', 'LOG_RESTORE_TOPIC', 'LOG_UNLOCK'];
```

Comment:
more like `noDetail`. Also `static` (and considering its usage, probably should be a `Set` instead of plain array).